pkg/keymanagementprovider/azurekeyvault/provider.go

```diff
@@ -177,17 +201,30 @@ func (s *akvKMProvider) GetKeys(ctx context.Context) (map[keymanagementprovider.
 			return nil, nil, fmt.Errorf("failed to get key objectName:%s, objectVersion:%s, error: %w", keyVaultKey.Name, keyVaultKey.Version, err)
 		}
 
+		if keyVaultKey.Version == "" {
+			keyVaultKey.Version = getObjectVersion(*keyBundle.Key.Kid)
+		}
+
 		if keyBundle.Attributes != nil && keyBundle.Attributes.Enabled != nil && !*keyBundle.Attributes.Enabled {
-			return nil, nil, fmt.Errorf("key %s version %s is disabled. please re-enable in azure key vault or remove reference to this key", keyVaultKey.Name, keyVaultKey.Version)
+			isEnabled := false
+			startTime := time.Now()
+			lastRefreshed := startTime.Format(time.RFC3339)
+
+			keysMap[keymanagementprovider.KMPMapKey{Name: keyVaultKey.Name, Version: keyVaultKey.Version, Enabled: isEnabled}] = nil
+			properties := getStatusProperty(keyVaultKey.Name, keyVaultKey.Version, strconv.FormatBool(isEnabled), lastRefreshed)
+			keysStatus = append(keysStatus, properties)
+
+			continue
```

Comment:
@akashsinghal do you think if it's a breaking change as we return the error before and continue now?

Reply:
Yeah it is a breaking change. However, I think the preferred behavior is the one @duffney is introducing here. Failing on error as we are doing currently is exponentially more disruptive than it was before considering we have refresh capability. I would argue in favor of this change and note as breaking change in the release note.